daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/ProcessorStateBases.scala

```diff
@@ -480,6 +480,17 @@ abstract class ParseOrUnparseState protected (
     res
   }
 
+  private var _hiddenDepth = 0
+
+  def incrementHiddenDef = {
+    _hiddenDepth += 1
+  }
+  def decrementHiddenDef = {
+    _hiddenDepth -= 1
+  }
+
+  def withinHiddenNest: Boolean = _hiddenDepth > 0
```

Comment:
Related to what Mike mentioned in the debugger, changing this variable to ``withinHiddenGroup`` is maybe more consistent with the DFDL spec.

Reply:
The word "nest" here is important. The concept here is that the runtime is beneath/inside a hidden group, but not directly/statically, but anywhere within the dynamic context nest. 

I'm open to other names, but the goal is to denote the dynamic extent of the hidden group.

Reply:
Fair enough. Should we modify the description of "info hidden" to mention "...within the nesting context of a hidden group" or something like that to be consistent, or is that distinction to verbose/excessive for the debugger?

Reply:
That would be good to do.